services/wireguard/endpoint/remoteclient/client.go

```diff
@@ -137,6 +142,27 @@ func (c *client) PeerStats(iface string) (*wg.Stats, error) {
 	}, nil
 }
 
-func (c *client) Close() error {
-	return c.wgc.Close()
+func (c *client) Close() (err error) {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+
+	var errs []error
+	defer func() {
+		for i := range errs {
+			log.Error().Err(errs[i]).Msg("Failed to close wireguard remote client")
+			if err == nil {
+				err = errs[i]
```

Comment:
Returning just last error, `ErrorComposite.Err()` would be useful to return multiple errors

Reply:
Or just wrap them all with %w